### PR TITLE
Add username Backend and Frontend in Signup Screen

### LINKED ISSUE #115: Add username to User Model

### PR DESCRIPTION
**Pull request by rita-naimi**
### What
Adds username selection to the signup flow. Users pick a unique username (3–20 characters, alphanumeric plus underscores, hyphens, and periods) with real-time availability checking.

Includes:
- Username text field on the Basic Info screen
- Real-time validation and availability checking (300ms debounce)
- Visual feedback: loading spinner, green checkmark when available, red X when taken
- `username` field added to `User` model with validation
- Repository methods to check username availability (Local + Firestore)
- ViewModel integration for username state management
- Tests covering validation, availability logic, and edge cases

### Why
Users need a unique identifier for their account.  
This enables:
- @mentions
- Profile links / public usernames
- Username-based search

Real-time availability feedback improves the signup experience by preventing errors at submission time.

### How
- **Model:** Added `username` to `User`, stored in lowercase, with validation (3–20 chars, `[a-z0-9._-]`).
- **UI:** Added `UsernameTextField` composable that validates format live and triggers debounced availability checks.
- **ViewModel:** SignUpViewModel now manages username state & emits loading/available/taken/error states.
- **Repository:** Added `isUsernameAvailable()` and implemented in Local & Firestore.
- **Signup Flow:** Updated BasicInfoScreen and SignUpOrchestrator to include the username in account creation.
- **Tests:** Added validation tests, availability tests, and fixed async timing in Android tests.

### Notes
- Usernames are stored and compared case-insensitive.
- Debounce avoids excessive backend requests.

Fixes #115 and #116

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/repository/UserRepository.kt [hunk 1]
```diff
@@ -165,4 +165,12 @@ interface UserRepository {
    * @return A list of event IDs that are marked as favorite by the user.
    */
   suspend fun getFavoriteEvents(userId: String): List<String>
+
+  /**
+   * Checks if a username is available (not already taken). Performs case-insensitive check.
+   *
+   * @param username The username to check (will be normalized to lowercase for comparison).
+   * @return true if the username is available, false if it's already taken.
+   */
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/repository/UserRepository.kt]: nice documentation!
